File: src/main/java/sk/tuke/kpi/BlockPuzzle/core/blocks/Block.java
package sk.tuke.kpi.BlockPuzzle.core.blocks;

import sk.tuke.kpi.BlockPuzzle.consoleui.ConsoleColor;
import sk.tuke.kpi.BlockPuzzle.core.Cell;

public class Block {
    private final Cell[][] shape;
    private final ConsoleColor color;

    public Block(Cell[][] shape, ConsoleColor color) {
        this.shape = shape;
        this.color = color;
    }

    public ConsoleColor getColor() {
        return color;
    }

    public Cell[][] getShape() {
        return shape;
    }

    public int getWidth() {
        return shape[0].length;
    }

    public int getHeight() {
        return shape.length;
    }
}

File: gamestudio-frontend/src/core/Block.js
class Block {
    constructor(shape, color) {
        this.shape = shape;
        this.color = color;
    }

    getWidth() {
        return this.shape[0].length;
    }

    getHeight() {
        return this.shape.length;
    }

    getShape() {
        return this.shape;
    }

    getColor() {
        return this.color;
    }
}

export default Block;